presto-main/src/test/java/io/prestosql/sql/planner/TestDomainTranslator.java

```diff
@@ -1582,6 +1582,30 @@ public void testLikePredicate()
         assertUnsupportedPredicate(not(like(C_VARCHAR, stringLiteral("abc\\_def"))));
     }
 
+    @Test
+    public void testStartsWithFunction()
+    {
+        Type varcharType = createUnboundedVarcharType();
+
+        // constant
+        testSimpleComparison(
```

Comment:
add some edge cases like:
* empty pattern.
* pattern with trailing spaces
* pattern with escaped `.*` or `/`

Reply:
Added empty pattern which is not supported in domain translation and `_` prefix pattern which is treated as a constant string as different from `LIKE`.

> - pattern with trailing spaces
> - pattern with escaped `.*` or `/`

What do these cases test? Unlike `LIKE` predicate, `STARTS_WITH` function always treats the second argument as  a constant string. It doesn't care whether characters are escaped or not.

Reply:
> What do these cases test? Unlike LIKE predicate, STARTS_WITH function always treats the second argument as a constant string. It doesn't care whether characters are escaped or not.

You are right.